runtime/frequency/src/lib.rs

```diff
@@ -1150,16 +1347,46 @@ const RELAY_CHAIN_SLOT_DURATION_MILLIS: u32 = 6_000;
 
 // See https://paritytech.github.io/substrate/master/pallet_parachain_system/index.html for
 // the descriptions of these configs.
-#[cfg(any(not(feature = "frequency-no-relay"), feature = "frequency-lint-check"))]
+#[cfg(any(
+	not(feature = "frequency-no-relay"),
+	feature = "frequency-lint-check",
+	feature = "frequency-bridging"
+))]
 impl cumulus_pallet_parachain_system::Config for Runtime {
 	type RuntimeEvent = RuntimeEvent;
 	type OnSystemEvent = ();
 	type SelfParaId = parachain_info::Pallet<Runtime>;
+
+	#[cfg(feature = "frequency-bridging")]
+	type DmpQueue = frame_support::traits::EnqueueWithOrigin<MessageQueue, RelayOrigin>;
+
+	#[cfg(not(feature = "frequency-bridging"))]
 	type DmpQueue = frame_support::traits::EnqueueWithOrigin<(), sp_core::ConstU8<0>>;
+
+	#[cfg(not(feature = "frequency-bridging"))]
 	type ReservedDmpWeight = ();
+
+	#[cfg(feature = "frequency-bridging")]
+	type ReservedDmpWeight = ReservedDmpWeight;
+
+	#[cfg(not(feature = "frequency-bridging"))]
 	type OutboundXcmpMessageSource = ();
+
+	#[cfg(feature = "frequency-bridging")]
+	type OutboundXcmpMessageSource = XcmpQueue;
+
+	#[cfg(not(feature = "frequency-bridging"))]
 	type XcmpMessageHandler = ();
+
+	#[cfg(feature = "frequency-bridging")]
+	type XcmpMessageHandler = XcmpQueue;
+
+	#[cfg(not(feature = "frequency-bridging"))]
 	type ReservedXcmpWeight = ();
+
+	#[cfg(feature = "frequency-bridging")]
+	type ReservedXcmpWeight = ReservedXcmpWeight;
+
```

Comment:
This gets really messy to read... might be better to `#[cfg(...)]` the entire trait implementation, ie:
```rust
#[cfg(feature = "frequency-bridging")]
impl cumulus_pallet_parachain_system::Config for Runtime { ... }
#[cfg(not(feature = "frequency-bridging"))]
impl cumulus_pallet_parachain_system::Config for Runtime { ... }
```